daffodil-core/src/main/scala/org/apache/daffodil/grammar/ElementBaseGrammarMixin.scala

```diff
@@ -625,6 +630,15 @@ trait ElementBaseGrammarMixin
     }
   }
 
+  private lazy val textStandardNonBaseTenConverter = {
+    primType match {
+      case PrimType.Double | PrimType.Float | PrimType.Decimal =>
```

Comment:
While this is shorter, you are depending on the set of types not changing here. E.g., if a new type is introduced (quad-precision floats?) then this will break. 

It is better to positively select the integer types that support this than use negation of the the floating point types that do not. 

You can still do this without enumerating all the integer types because they're all subtypes of PrimType.Integer.Kind 

`case _: PrimType.Integer.Kind => ...`

Reply:
Will do, I'll also fix up Zoned with a similar logic while I'm there.